### PR TITLE
Convert headless tables to the summary list component

### PR DESCRIPTION
**Pull request by kr8n3r**
Migrate headless tables to a govuk-summary-list component as that's the more accessible way of presenting it to the user for interaction.

See individual commits for details. Fixes https://trello.com/c/7I9xjoqU/740-change-tables-without-column-headers-to-use-the-govuk-frontend-summary-list-pattern

Related: https://github.com/alphagov/notifications-functional-tests/pull/505

<details>
  <summary>Service settings</summary>

**Before**
service-settings-before

**After**
service-settings-after

**Mobile before**
service-seetings-mobile-before

**Mobile after**
service-settings-mobile

</details>

<details>
  <summary>Organisation settings</summary>

**Before**
org-settings-before

**After**
org-settings-after

**Mobile before**
org-settings-before-mobile

**Mobile after**
org-settings-mobile

</details>

<details>
  <summary>Data retention settings</summary>

**Before**
data-retention-before

**After**
data-retention-after

**Mobile before**
data-retention-mobile-before

**Mobile after**
data-retention-mobile

</details>

### CODE REVIEW

**Comment by kr8n3r** [comment]
> Can you make sure that the widths of the columns still match the grid in #2931 please and thanks

certainly 
Screenshot 2024-05-15 at 15 59 34

**Review by tombye — CHANGES_REQUESTED** [review]
I spotted a few things that I've added comments about.

I think the `<dt>` needs to be `35%` including padding, based on [Chris's previous work on the settings pages](https://github.com/alphagov/notifications-admin/pull/2931/files) so we'll have to introduce a new class like `notify-summary-list__key--one-quarter`.

If all that can be sorted, this should be good to go. 👏🏻 👏🏻 for doing such a large piece of work that'll tick off a need that keeps coming up in accessibility audits 🏅.

File: app/templates/views/service-settings.html [hunk 1]
```diff
@@ -1,304 +1,472 @@
...
+                  "visuallyHiddenText": "sending files by email",
+                  "classes": "govuk-link--no-visited-state"
+                }
+              ]
+            }
+          }
+        ]%}
+      {% endif %}
+
+      {% set email_settings_rows = base_email_settings_row + restricted_email_settings_rows  if restricted_email_settings_rows else base_email_settings_row %}
```
[the remark this answers is not in the document]
Comment (kr8n3r) [reply app/templates/views/service-settings.html]: sorted

File: app/templates/views/service-settings.html [hunk 2]
```diff
@@ -1,304 +1,472 @@
...
+      {% set text_message_settings_rows = base_text_message_setting_row + restricted_text_message_settings_rows  if restricted_text_message_settings_rows else base_text_message_setting_row %}
+
+      {{ govukSummaryList({
+        "classes": "notify-summary-list service-sms-settings",
+        "rows": text_message_settings_rows
+      }) }}
+    {% endset %}
+
+    {% set service_letter_settings %}
+      {% set base_etter_settings_row = [
```
[the remark this answers is not in the document]
Comment (kr8n3r) [reply app/templates/views/service-settings.html]: sorted, and fixed an additional one as well. I had camelCase before so had to rename them all

**Comment by kr8n3r** [comment]
> I think the `<dt>` needs to be `35%` including padding, based on [Chris's previous work on the settings pages](https://github.com/alphagov/notifications-admin/pull/2931/files) so we'll have to introduce a new class like `notify-summary-list__key--one-quarter`.
> 

yeah, i think i was reading the table layout wrongly as I saw 2/8

**Comment by kr8n3r** [comment]
I have also applied the key width class to the /user-settings table as well

**Review by tombye — APPROVED** [review]
Thanks for making the changes, and plugging them into the user profile page.

This looks good to go now 👍🏻

**Pull request merged**